### ISSUE TITLE
`CrossReferencing()` fails with `object 'my.lib' not found` when loading `compound_db.qs`

### ISSUE THREAD

**Issue by hosseinvk** [xia-lab/MetaboAnalystR#375]
## Summary

I am using **MetaboAnalystR 4.2.0** on a Unix/Linux server with **R 4.5.1**. Running `CrossReferencing(mSet, "name")` fails when loading `compound_db.qs`.

## Error

```r
mSet <- CrossReferencing(mSet, "name")
```

Output:

```text
[1] "Reading data unsuccessful, attempting to re-download file..."
100 1009k  100 1009k    0     0   421k      0  0:00:02  0:00:02 --:--:--  421k
[1] "Loading files from server unsuccessful. Ensure curl is downloaded on your computer."
Error in .get.my.lib("compound_db.qs") : object 'my.lib' not found
```

## Checks performed

The downloaded file exists and has a plausible size:

```r
file.exists("compound_db.qs")
file.info("compound_db.qs")$size
```

Output:

```text
[1] TRUE
[1] 1033560
```

But it cannot be read with `qs::qread()`:

```r
packageVersion("qs")
packageVersion("stringfish")
qs::qread("compound_db.qs")
```

Output:

```text
[1] '0.25.5'
[1] '0.18.0'

Error in qs::qread("compound_db.qs") : QS format not detected
```

I also tested `qs 0.27.3`, with the same `QS format not detected` error.

The file appears to be binary, not an HTML/text error page:

```r
readBin("compound_db.qs", "raw", n = 20)
```

Output:

```text
[1] 0b 0e 0a c1 01 01 02 01 00 00 00 00 00 00 00 00 55 b7 fb 5a
```

## Environment

```text
R version 4.5.1
Platform: x86_64-pc-linux-gnu
MetaboAnalystR 4.2.0
qs 0.25.5
stringfish 0.18.0
```

## Question

Is the hosted `compound_db.qs` expected to be readable with `qs::qread()`? It seems `.get.my.lib()` retries the download after `qread()` fails, but then returns `my.lib` even though it was never created, causing the final `object 'my.lib' not found` error.

**Comment by hosseinvk** [comment]
Dear @Zhiqiang-PANG,

Thank you for your recent work on the application and for replacing `qs2` with `qs`. The original installation issue is now resolved.

However, further downstream I am now facing a new error when running `CalculateHyperScore()`.

The following steps run successfully, but the final command fails:

```r
packageVersion("MetaboAnalystR")
# [1] ‘4.3.0’

mSet <- InitDataObjects("conc", "msetora", FALSE, 150) # OK

cmpd.vec<-c("Auricularine", "Xylose", "Hydroquinone" ...)

mSet <- Setup.MapData(mSet, cmpd.vec) # OK
# Warning message:
# In write.csv(qvec, file = "datalist.csv", row.names = FALSE, col.names = FALSE, :
#   attempt to set 'col.names' ignored

mSet <- CrossReferencing(mSet, "name") # OK
# [1] "compound_db.qs"
# [1] "Loaded files from MetaboAnalyst web-server."
# [1] "syn_nms.qs"
# [1] "Loaded files from MetaboAnalyst web-server."
# [1] "2"
# [2] "There are >15 compounds without matches. You can either proceed or if necessary, update these compound IDs and upload again."

mSet <- CreateMappingResultTable(mSet) # OK
# [1] "compound_db.qs"
# [1] "Loaded files from MetaboAnalyst web-server."

mSet <- Setup.HMDBReferenceMetabolome(
  mSet,
  file.path(model_dir, "tensor", "background_FE", "bg_metabolites_CB.txt")
) # OK
# Read 1190 items
# [1] "master_compound_db.qs"
# [1] "Loaded files from MetaboAnalyst web-server."

mSet <- SetMetabolomeFilter(mSet, TRUE) # OK
mSet <- SetCurrentMsetLib(mSet, "smpdb_pathway", 2) # OK

mSet <- CalculateHyperScore(mSet) # ERROR
# [1] "compound_db.qs"
# [1] "Loaded files from MetaboAnalyst web-server."
# Error in mSetObj$analSet : $ operator is invalid for atomic vectors
# In addition: Warning message:
# In value[[3L]](cond) :
#   ExportResultMatArrow failed: cannot connect to localhost:6311
```

I am sharing this update in case it helps with tracking the current state of the issue.

Many thanks.
